### PR TITLE
OneSignal packages

### PR DESCRIPTION
**Pull request by stan-osipov**
## Purpose of this PR
OneSignal Unity SKD is split into 3 packages

* `com.onesignal.unity.core`
* `com.onesignal.unity.ios`
* `com.onesignal.unity.android`

### com.onesignal.unity.core
Core package Contains C# API, Samples, Settings window UI

#### C# API Changes
* Main OneSignal API access point moved to the `OneSignalPush` namespace.
* Windows Phone platform related code removed. 
* Models moved to the separate files.
* To avoid breaking changes due to `OneSignalPush` namespace the `public class OneSignal : OneSignalPush.OneSignal` class added.
* Only Runtime `OneSignalPush.asmdef` is auto referenced. Any other `*.asmdef`s  in this and related packages are not.

#### Samples
Unmodified sample from the original OneSignal Unity SDK.

#### Settings window UI
First Settings window version that allows setting the application id.
Settings `ScriptableObject` is located under Assets/Plugins/OneSignal. Settings object is not included in the package, and the user can move settings object to any preferred location if needed.

### com.onesignal.unity.ios
* Contain iOS platform client.
* Only Included in iOS platform build.

### com.onesignal.unity.android
* Contain Android platform client.
* Only Included in Android platform build.
* Package will automatically add Google Packages Registry to the project `manifest.json` 

```
"name": "Game Package Registry by Google",
"url": "https://unityregistry-pa.googleapis.com",
"scopes": [
    "com.google"
]
```

And  will declare the `com.google.external-dependency-manager` dependency in own `package.json`

### IMGUI
The core package has `com.unity.modules.imgui@1.0.0` dependency, but **Unity 2018**  is not able to resolve it.  No compilation complains from Package Manager API it just won't do anything. So if user does not have this dependency declared in project `manifest.json` (very unlikely but still) we will get the compilation errors.

### ANDROID JNI
The `com.unity.modules.androidjni` is not recognized by the **Unity 2018**. Therefore we can't have it in the android package. There is a chance if the user will remove this default dependency manually from the `manifest.json` we will have the compilation errors.

## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/299)

### CODE REVIEW

File: Packages/com.onesignal.unity.core/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "author": {
+    "name": "OneSignal",
+    "email": "support@onesignal.com",
+    "url": "https://onesignal.com/"
+  },
+
+  "homepage": "https://onesignal.com/",
+  "bugs": {
+    "url": "https://github.com/OneSignal/OneSignal-Unity-SDK/issues",
+    "email" : "support@stansassets.com"
```
Comment (jkasten2) [inline Packages/com.onesignal.unity.core/package.json:29]: Should be `support@onesignal.com`

File: Packages/com.onesignal.unity.ios/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   "author": {
+     "name": "OneSignal",
+     "email": "support@onesignal.com",
+     "url": "https://onesignal.com/"
+   },
+
+   "homepage": "https://onesignal.com/",
+   "bugs": {
+     "url": "https://github.com/OneSignal/OneSignal-Unity-SDK/issues",
+     "email" : "support@stansassets.com"
```
Comment (jkasten2) [inline Packages/com.onesignal.unity.ios/package.json:30]: Should be `support@onesignal.com`

File: Packages/com.onesignal.unity.android/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  "license": "MIT",
+  "author": {
+    "name": "OneSignal",
+    "email": "support@onesignal.com",
+    "url": "https://onesignal.com/"
+  },
+  "homepage": "https://onesignal.com/",
+  "bugs": {
+    "url": "https://github.com/OneSignal/OneSignal-Unity-SDK/issues",
+    "email": "support@stansassets.com"
```
Comment (jkasten2) [inline Packages/com.onesignal.unity.android/package.json:29]: Should be support@onesignal.com

File: Packages/com.onesignal.unity.android/Editor/OneSignalEditorScript_Android.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+   const string k_AndroidConfigFolder = "Packages/com.onesignal.unity.android/Plugins/Android/OneSignalConfig";
+   const string k_PackageManifestPath = "Packages/com.onesignal.unity.android/package.json";
+
+   static OneSignalEditorScriptAndroid()
+   {
+      CreateOneSignalAndroidManifest();
+      PackageManifestSanityCheck();
+   }
+
+   static void PackageManifestSanityCheck()
```
[the remark this answers is not in the document]
Comment (stan-osipov) [reply Packages/com.onesignal.unity.android/Editor/OneSignalEditorScript_Android.cs:44]: sure.

File: Packages/com.onesignal.unity.android/Runtime/OneSignalAndroid.cs [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+using UnityEngine;
+using System.Collections.Generic;
+using System;
+using OneSignalPush.MiniJSON;
+
+namespace OneSignalPush.Android
```
[the remark this answers is not in the document]
Comment (stan-osipov) [reply Packages/com.onesignal.unity.android/Runtime/OneSignalAndroid.cs:33]: will do!

File: Packages/com.onesignal.unity.core/Editor/Core/ScopeRegistriesConfig.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+        public static ScopeRegistry GoogleScopeRegistry =>
+            new ScopeRegistry("Game Package Registry by Google",
+                GoogleScopeRegistryUrl,
+                new HashSet<string>
+                {
+                    "com.google"
+                });
+
+        public static ScopeRegistry NpmjsScopeRegistry =>
```
[the remark this answers is not in the document]
Comment (stan-osipov) [reply Packages/com.onesignal.unity.core/Editor/Core/ScopeRegistriesConfig.cs:24]: no, it's not. Same copy-paste mistake.

File: Packages/com.onesignal.unity.core/Editor/SettingsWindow/OneSignalSettingsWindow.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
+using UnityEditor;
+using UnityEngine;
+
+namespace OneSignalPush.Editor
+{
+    class OneSignalSettingsWindow :  OneSignalBaseSettingsWindow<OneSignalSettingsWindow>
+    {
+        const string k_SdkSetupUrl = "https://documentation.onesignal.com/docs/unity-sdk-setup";
+        const string k_HeaderText = "OneSignal is the market leader in customer engagement, powering mobile push, web push, email, and in-app messages.";
```
[the remark this answers is not in the document]
Comment (stan-osipov) [reply Packages/com.onesignal.unity.core/Editor/SettingsWindow/OneSignalSettingsWindow.cs:9]: Sure, but I am pretty sure we would like to make another iteration on the About tab.

File: Packages/com.onesignal.unity.core/Runtime/AssemblyInfo.cs [hunk 8]
```diff
@@ -0,0 +1,6 @@
+﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("OneSignalPush.Editor")]
+[assembly: InternalsVisibleTo("OneSignalPush.iOS")]
+[assembly: InternalsVisibleTo("OneSignalPush.Android")]
+[assembly: InternalsVisibleTo("OneSignalPush.WindowsPhone")]
```
[the remark this answers is not in the document]
Comment (stan-osipov) [reply Packages/com.onesignal.unity.core/Runtime/AssemblyInfo.cs:6]: yep, you right we were talking about it, I remove package but forgot the directive.

File: Packages/com.onesignal.unity.core/Runtime/Core/OneSignalLegacy.cs [hunk 9]
```diff
@@ -0,0 +1,5 @@
+public class OneSignal : OneSignalPush.OneSignal
```
[the remark this answers is not in the document]
Comment (stan-osipov) [reply Packages/com.onesignal.unity.core/Runtime/Core/OneSignalLegacy.cs:1]: Sure, I will also add the deprecation warning.

**Pull request closed without merging**